contactsummary/tests/phpunit/CRM/Contactsummary/Utils/ContractSummaryTestTrait.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * A trait with helper methods to be reused among this extension's tests
+ */
+trait ContractSummaryTestTrait {
+
+  /**
+   * Creates absence period from specified start and end dates and return its ID
+   *
+   * @param $startDate
+   * @param $endDate
+   *
+   * @return int
+   */
+  protected function createAbsencePeriod($startDate, $endDate)  {
+    $params = array('name'=> $startDate,'start_date'=>$startDate, 'end_date'=>$endDate);
+    $period = civicrm_api3('HRAbsencePeriod', 'create', $params);
+    return $period['id'];
+  }
+
+  /**
+   * Creates absence entitlements for specific leave type
+   *
+   * @param $params
+   *
+   * @return int
+   */
+  protected function createAbsenceEntitlement($params)  {
+    $period = CRM_HRAbsence_BAO_HRAbsenceEntitlement::create($params);
```

Comment:
That's what I initially did , but creating an absence period  using BAO was throwing an error.. the date was converted magically to year only format , so 2016-06-20 become 2016 which cause the error ... I investigated it for a while with no luck that's why I just used the API to avoid wasting more time on it.